src/middleware/nonce2-tx-middleware.ts

```diff
@@ -0,0 +1,69 @@
+import debug from 'debug'
+import { NonceTx, Transaction, MessageTx } from '../proto/loom_pb'
+import { ITxMiddlewareHandler, Client, ITxResults, AccountType } from '../client'
+import { Address } from '../address'
+
+const log = debug('nonce2-tx-middleware')
+
+export const INVALID_TX_NONCE_ERROR = 'Invalid tx nonce'
+
+export function isInvalidTxNonceError(err: any): boolean {
+  return err instanceof Error && err.message === INVALID_TX_NONCE_ERROR
+}
+
+/**
+ * Wraps data in a NonceTx.
+ * This middleware obtains the latest nonce from the chain for each tx.
+ * The Loom DAppChain keeps track of the nonce of the last committed tx to prevent replay attacks.
+ */
+export class Nonce2TxMiddleware implements ITxMiddlewareHandler {
```

Comment:
`Nonce2TxMiddleware` shouldn't be needed anymore, overload the constructors of the existing nonce middleware like so:

```typescript
constructor(publicKey: Uint8Array, client: Client)
constructor(account: Address, client: Client)
constructor(publicKeyOrAccount: Uint8Array | Address, client: Client) {
  if publicKeyOrAccount instanceof Address {
    // use account to get nonce
  } else {
    // use public key to get nonce
  }
}
```